client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx

```diff
@@ -462,6 +429,44 @@ class EditContactInfoFormCard extends Component {
 		return this.state.formSubmitting || includes( unmodifiableFields, snakeCase( name ) );
 	};
 
+	updateWpcomEmail( newContactDetails, updateWpcomEmail ) {
+		const { email } = newContactDetails;
+		if ( updateWpcomEmail && email && this.props.currentUser.email !== email ) {
+			wp.me()
+				.settings()
+				.update( { user_email: email } )
+				.then( ( data ) => {
+					if ( data.user_email_change_pending ) {
+						this.props.infoNotice(
+							this.props.translate(
+								'There is a pending change of your WordPress.com email to %(newEmail)s. Please check your inbox for a confirmation link.',
+								{
+									args: { newEmail: data.new_user_email },
+								}
+							),
+							{
+								showDismiss: true,
+								isPersistent: true,
+								duration: null,
+							}
+						);
+					}
+				} )
+				.catch( () => {
+					this.props.errorNotice(
+						this.props.translate(
+							'There was a problem updating your WordPress.com Account email.'
+						),
+						{
+							showDismiss: true,
+							isPersistent: true,
+							duration: null,
+						}
+					);
+				} );
```

Comment:
I'd love if we could move this logic outside of the component and just call the thunk from this method (e.g., `this.props.updateWpcomEmail` and then everything is handled through Redux.)